src/modules/Dropdown/Dropdown.js

```diff
@@ -82,121 +66,591 @@ export default class Dropdown extends Component {
+    floating: PropTypes.bool,
+
+    inline: PropTypes.bool,
+    labeled: PropTypes.bool,
+    linkItem: PropTypes.bool,
+
+    /** Allow selecting multiple options. */
+    multiple: PropTypes.bool,
+
+    /** Use a detached menu that is pointing to the Dropdown. */
+    pointing: PropTypes.oneOfType([
+      PropTypes.bool,
+      PropTypes.oneOf(_meta.props.pointing),
+    ]),
+
+    /** Make the dropdown options searchable. */
+    search: PropTypes.bool,
+
+    /** The text displayed in the dropdown, usually for the active item. */
+    text: PropTypes.string,
 
-  static defaultProps = {
-    icon: 'dropdown',
+    // TODO 'searchInMenu' or 'search='in menu' or ???  How to handle this markup and functionality?
+
+    /** Behave as an html select. */
+    selection: PropTypes.bool,
+    simple: PropTypes.bool,
+
+    loading: PropTypes.bool,
+    error: PropTypes.bool,
+    disabled: PropTypes.bool,
+
+    scrolling: PropTypes.bool,
   }
 
-  componentDidMount() {
-    this.element = $(this.refs.element).dropdown({
-      ...getPluginProps(this.props, pluginPropTypes),
-      onAdd: this.handleAdd,
-      onChange: this.handleChange,
-      onRemove: this.handleRemove,
+  static autoControlledProps = [
+    'open',
+    'value',
+  ]
+
+  static _meta = _meta
+  static Divider = DropdownDivider
+  static Item = DropdownItem
+  static Menu = DropdownMenu
+
+  componentWillMount() {
+    if (super.componentWillMount) super.componentWillMount()
+    // console.debug('Dropdown.componentWillMount()')
+    const { open, options } = this.props
+    const { value } = this.state
+
+    const selectedIndex = this.getItemIndexByValue(value || _.get(options, '[0].value'))
+
+    this.trySetState({
+      value,
+    }, {
+      selectedIndex,
     })
+
+    if (open) this.open()
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    _.invoke(this, 'element.dropdown', 'refresh')
-    const isDOMInSync = _.isEqual(this.props.value, this.getValue())
-    if (!isDOMInSync && this.isSelection()) this._syncFromProps()
+  shouldComponentUpdate(nextProps, nextState) {
+    return !_.isEqual(nextProps, this.props) || !_.isEqual(nextState, this.state)
   }
 
-  componentWillUnmount() {
-    _.invoke(this, 'element.off')
+  componentWillReceiveProps(nextProps) {
+    super.componentWillReceiveProps(nextProps)
+    // console.debug('Dropdown.componentWillReceiveProps()')
+    // console.log('changed props:', _.transform(nextProps, (res, val, key) => {
+    //   if (val !== this.props[key]) res[key] = val
+    // }, {}))
+
+    if (nextProps.value !== this.state.value) {
+      this.setValue(nextProps.value)
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) { // eslint-disable-line complexity
+    // console.debug('Dropdown.componentDidUpdate()')
+    // console.log('changed state:', _.transform(this.state, (res, val, key) => {
+    //   if (val !== prevState[key]) res[key] = val
+    // }, {}))
```

Comment:
Yep, from the description:

>### Areas of concern
>These are the areas to watch and comment on. These are some known ugly parts. They've been added as TODOs to the main PR #206.
>
>### Debug / Logging
>
>Currently, nearly every method has a console.debug() when called and a console.log() for any relevant debugging data. This would be pulled into an actual debug util. Consider:
>
>- debug module
>- removal in production code
>- runtime flag to enable/disable by module
>...